tests/unit/dcc_comms/test_mqtt_dcc_comms.py

```diff
@@ -0,0 +1,193 @@
+#  - Redistributions in binary form must reproduce the above copyright        #
+#      notice, this list of conditions and the following disclaimer in the    #
+#      documentation and/or other materials provided with the distribution.   #
+#                                                                             #
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"#
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE  #
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE #
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE  #
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR        #
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF       #
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS   #
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+import unittest
+
+import mock
+
+from liota.dcc_comms.mqtt_dcc_comms import MqttDccComms
+from liota.entities.edge_systems.dell5k_edge_system import Dell5KEdgeSystem
+from liota.lib.transports.mqtt import Mqtt, MqttMessagingAttributes
+
+
+class MqttDccCommsTest(unittest.TestCase):
+    """
+    Unit test cases for MQTT DccComms
+    """
+
+    @mock.patch.object(Mqtt, 'connect_soc')
+    def setUp(self, mock_connect):
+        """
+        Setup all required parameters for MQTT DCC communication tests
+        :param mock_connect: Mocked MQTT connect method
+        :return:
+        """
+        # Create Edge System
+        self.edge_system = Dell5KEdgeSystem("TestGateway")
+
+        self.url = "Broker-IP"
+        self.port = "Broker-Port"
+        self.clean_session = True
+        self.publish_message = "test-message"
+
+        # Creating messaging attributes
+        self.mqtt_msg_attr = MqttMessagingAttributes(pub_topic="publish-topic")
+
+        # Instantiate client for DCC communication
+        self.client = MqttDccComms(edge_system_name=self.edge_system.name,
+                                   url=self.url, port=self.port, clean_session=self.clean_session)
+
+    def tearDown(self):
+        """
+        Clean up all parameters used in test cases
+        :return:
+        """
+        self.edge_system = None
+        self.url = None
+        self.port = None
+        self.clean_session = None
+        self.publish_message = None
+        self.mqtt_msg_attr = None
+        self.client = None
+
+    @mock.patch.object(Mqtt, 'connect_soc')
+    def test_init_no_msgattr_no_client_id(self, mock_connect):
```

Comment:
Test cases for `init()` should be re-written to check the following:

- MqttDccComms called with edge_system_name, msg_attr=None and no client id.  Here test for client_id is auto-generated and is the edge_system_uuid and whether everything is written correctly to `uuid.ini` file
- MqttDccComms called with edge_system_name, msg_attr=None and client id.  Here test for client_id is user-provided and whether everything is written correctly to `uuid.ini` file


- MqttDccComms called with edge_system_name, msg_attr=valid msg_attr and no client id.  Here test for client_id is `""`
- MqttDccComms called with edge_system_name, msg_attr=valid msg_attr  and client id.  Here test for client_id is user-provided